sympy/combinatorics/rewritingsystem.py

```diff
@@ -285,3 +298,190 @@ def reduce(self, word, exclude=None):
                 if new != prev:
                     again = True
         return new
+
+    def _compute_inverse_rules(self, rules):
+        '''
+        Compute the inverse rules for a given set of rules.
+        The inverse rules are used in the automaton for word reduction.
+
+        Arguments:
+            rules (dictionary): Rules for which the inverse rules are to computed.
+
+        Returns:
+            Dictionary of inverse_rules.
+
+        '''
+        inverse_rules = {}
+        for r in rules:
+            rule_key_inverse = r**-1
+            rule_value_inverse = (rules[r])**-1
+            if (rule_value_inverse < rule_key_inverse):
+                inverse_rules[rule_key_inverse] = rule_value_inverse
+            else:
+                inverse_rules[rule_value_inverse] = rule_key_inverse
+        return inverse_rules
+
+    def construct_automaton(self):
+        '''
+        Construct the automaton based on the set of reduction rules of the system.
+
+        Automata Design:
+        The accept states of the automaton are the proper prefixes of the left hand side of the rules.
+        The complete left hand side of the rules are the dead states of the automaton.
+
+        '''
+        # Create a finite state machine as an instance of the StateMachine object
+        fsm = StateMachine('fsm')
+        fsm.add_state('start', state_type='s')
+
+        self.reduction_automaton = fsm
```

Comment:
This could just be `self.reduction_automaton = StateMachine('fsm')` if you add the start state in the `StateMachine` class.